File: resources/sdk/webmessagingjava/extensions/ApiResponse.java
package cloud.genesys.webmessaging.sdk;

import java.util.Map;

public interface ApiResponse<T> extends AutoCloseable {
    Exception getException();
    int getStatusCode();
    String getStatusReasonPhrase();
    boolean hasRawBody();
    String getRawBody();
    T getBody();
    Map<String, String> getHeaders();
    String getHeader(String key);
    String getCorrelationId();
}

File: resources/sdk/purecloudkotlin/extensions/ApiResponse.kt
package com.mypurecloud.sdk.v2

import java.lang.Exception

interface ApiResponse<T> : AutoCloseable {
    val exception: Exception?
    val statusCode: Int
    fun hasRawBody(): Boolean
    val rawBody: String?
    val pBody: T
    fun getBody() : T?
    val headers: Map<String?, String?>
    fun getHeader(key: String?): String?
    fun getCorrelationId() : String?
    fun getStatusReasonPhrase() : String?
}
